agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/reminder/ReminderDetailScreen.kt

```diff
@@ -0,0 +1,220 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.pronoidsoftware.agenda.presentation.detail.reminder
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.pronoidsoftware.agenda.presentation.R
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailActionText
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailDescription
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailNotification
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailTime
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailTitle
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailToolbar
+import com.pronoidsoftware.agenda.presentation.detail.components.AgendaDetailType
+import com.pronoidsoftware.core.presentation.designsystem.LocalSpacing
+import com.pronoidsoftware.core.presentation.designsystem.TaskyGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyLightGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+import com.pronoidsoftware.core.presentation.designsystem.TaskyWhite2
+import com.pronoidsoftware.core.presentation.designsystem.components.TaskyScaffold
+import com.pronoidsoftware.core.presentation.ui.ObserveAsEvents
+import com.pronoidsoftware.core.presentation.ui.formatFullDate
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+
+@Composable
+fun ReminderDetailScreenRoot(
+    onCloseClick: () -> Unit,
+    viewModel: ReminderDetailViewModel = hiltViewModel(),
+) {
```

Comment:
One of the tougher decisions to make for this project is whether to use one screen per agenda item type or one screen for all agenda items. 2 years ago, I was also team single screen per type, but my mind has changed :D

My concern was that having a single screen makes it harder to read and update, but it actually got a lot easier:
If you look at the UI hierarchy of the items, reminder and task screens are almost identical while the event screen also has the same elements with a few extra composables. 

The complete root screen structure is no more than this:
```
HeaderSection()
TitleSection()
DescriptionSection()
if(item is Event) {
    PhotoSection()
}
StartDateSection()
if(item is Event) {
    EndDateSection()
}
ReminderSection()
if(item is Event) {
    AttendeeSection()
}
DeleteButton()
```
As long as you extract the UI into single components (which you already did to a large degree), it remains very readable. Same goes for the underlying ViewModel, you will save so much code by just having one ViewModel, since the logic of what all items have in common only needs to be written once. And if you'd extend that screen with another type of agenda item, that would also be easier and wouldn't require you to duplicate most of this to another screen.

Reply:
Makes sense, and I was going back and forth with this decision too 😅 I was going to work on Reminder screen first at least and was thinking of a lot of copy and paste, but less code would be much nicer. I'll give it a try after I get the functionality of the Reminder detail screen working 👍